### ISSUE TITLE
Replace pypdf with pyHanko

### ISSUE THREAD

**Issue by bgermann** [xhtml2pdf/xhtml2pdf#755]
### Motivation
Both pypdf and pyHanko are dependencies. pyHanko includes a fork of PyPDF2, which is itself a fork of pypdf.

### Proposed Solution
The pyHanko variants of the used pypdf objects (PdfReader, PdfWriter, PageObject) should be used.

### Additional Context
Implementing this proposal enables to drop the pypdf dependency.

**Comment by stefan6419846** [comment]
> PyPDF2, which is itself a fork of pypdf

This is wrong for some time now: PyPDF2 has been deprecated in favor of pypdf. pyHanko does indeed include a heavily modified version of PyPDF2.

In my personal opinion, https://github.com/xhtml2pdf/xhtml2pdf/commit/6777fe9b1106ebb1d97ee1eb8dd624a0848a1b28 should be mostly reverted and pyHanko dropped completely for a clean separation of concerns.
